### ISSUE TITLE
"[NLP] RT  33.33.33.0/24 add failed-rt exists" is non-fatal

### ISSUE THREAD

**Issue by luisgerhorst** [loxilb-io/loxilb#442]
https://github.com/loxilb-io/loxilb/blob/8ea1f0e075b9ee08676208ed10b5b28983b54814/api/loxinlp/nlp.go#L1122

On a custom 6.5.0-rc6 kernel with Debian 11 (details upon request), I am triggering this error when running https://github.com/loxilb-io/loxilb/tree/main/cicd/tcplb locally:

```
...
DBG:  2023/11/14 13:13:16 rt added - 10.10.10.254/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  10.10.10.254/32 added
ERR:  2023/11/14 13:13:16 rt add - 10.10.10.0/24:root exists
ERR:  2023/11/14 13:13:16 [NLP] RT  10.10.10.0/24 add failed-rt exists
DBG:  2023/11/14 13:13:16 rt added - 10.10.10.255/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  10.10.10.255/32 added
DBG:  2023/11/14 13:13:16 rt added - 31.31.31.254/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  31.31.31.254/32 added
ERR:  2023/11/14 13:13:16 rt add - 31.31.31.0/24:root exists
ERR:  2023/11/14 13:13:16 [NLP] RT  31.31.31.0/24 add failed-rt exists
DBG:  2023/11/14 13:13:16 rt added - 31.31.31.255/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  31.31.31.255/32 added
DBG:  2023/11/14 13:13:16 rt added - 32.32.32.254/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  32.32.32.254/32 added
ERR:  2023/11/14 13:13:16 rt add - 32.32.32.0/24:root exists
ERR:  2023/11/14 13:13:16 [NLP] RT  32.32.32.0/24 add failed-rt exists
DBG:  2023/11/14 13:13:16 rt added - 32.32.32.255/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  32.32.32.255/32 added
DBG:  2023/11/14 13:13:16 rt added - 33.33.33.254/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  33.33.33.254/32 added
ERR:  2023/11/14 13:13:16 rt add - 33.33.33.0/24:root exists
ERR:  2023/11/14 13:13:16 [NLP] RT  33.33.33.0/24 add failed-rt exists
DBG:  2023/11/14 13:13:16 rt added - 33.33.33.255/32:root
INFO: 2023/11/14 13:13:16 [NLP] RT  33.33.33.255/32 added
ERR:  2023/11/14 13:13:16 rt add - 10.10.10.3/32:root exists
ERR:  2023/11/14 13:13:16 [NLP] RT  10.10.10.3/32 add failed-rt exists
DBG:  2023/11/14 13:13:16 ebpf intfmap added - 8 vlan 0 -> 11
DBG:  2023/11/14 13:13:16 ebpf txintfmap added - 11 -> 8
...
```

Thereafter, loxicmd reports the lb but it does not appear to 'listen' on the desired port. However, the HTTP servers are reachable from the loxilb container. Should this error really be non-fatal?

As the error suggests, adding a call to get route before create_lb_rule in config.sh shows existing routes. I still have to figure out why these exist.

```
...
+ sleep 5
+ docker exec -i llb1 loxicmd get route
| DESTINATIONIPNET |  GATEWAY   |   FLAG   |
|------------------|------------|----------|
| 0.0.0.0/0        | 172.17.0.1 | Ind      |
| 10.10.10.0/24    |            | Self     |
| 10.10.10.254/32  |            | Self     |
| 10.10.10.255/32  |            | Self     |
| 10.10.10.3/32    |            | Self     |
| 127.0.0.0/8      |            | Self     |
| 172.17.0.0/16    |            | Self     |
| 172.17.0.1/32    | 172.17.0.1 | Ind Host |
| 31.31.31.0/24    |            | Self     |
| 31.31.31.254/32  |            | Self     |
| 31.31.31.255/32  |            | Self     |
| 32.32.32.0/24    |            | Self     |
| 32.32.32.254/32  |            | Self     |
| 32.32.32.255/32  |            | Self     |
| 33.33.33.0/24    |            | Self     |
| 33.33.33.254/32  |            | Self     |
| 33.33.33.255/32  |            | Self     |
...
+ create_lb_rule llb1 20.20.20.1 --tcp=2020:8080 --endpoints=31.31.31.1:1,32.32.32.1:1,33.33.33.1:1
...
```

**Comment by nik-netlox** [comment]
Hi @luisgerhorst , You can ignore those error logs. They just means that the route that you are trying to add already exists. Second, loxilb is not a proxy load balancer so it won't listen to the port you specify in create lb rule. Lastly, the routes you are seeing are fetched from the kernel. These self routes are added when interfaces are created.

**Comment by luisgerhorst** [comment]
Thanks. 

> Second, loxilb is not a proxy load balancer so it won't listen to the port you specify in create lb rule.

Sorry, this was not clear. I meant it does not forward the connections to the targets.

I noticed it works on Debian Bullseye with a 5.10.0-26-amd64 kernel and on Ubuntu 22.04 with the 5.15 kernel, but not with the 6.5.0-rc6 I built using the Ubuntu jammy config. I will check newer kernel from upstream to see if it is a problem with my build or an incompatibility with newer kernels.
